wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/brigadier/BrigadierDsl.kt

```diff
@@ -0,0 +1,239 @@
+    val argument = RequiredArgumentBuilder.argument<S, Boolean>(
+        name,
+        BoolArgumentType.bool()
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a double with [name] as the parameter name.
+ * @param min the minimum value.
+ * @param max the maximum value.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.double(
+    name: String,
+    min: Double = -Double.MAX_VALUE,
+    max: Double = Double.MAX_VALUE,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, Double>(
+        name,
+        DoubleArgumentType.doubleArg(min, max)
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a float with [name] as the parameter name.
+ * @param min the minimum value.
+ * @param max the maximum value.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.float(
+    name: String,
+    min: Float = -Float.MAX_VALUE,
+    max: Float = Float.MAX_VALUE,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, Float>(
+        name,
+        FloatArgumentType.floatArg(min, max)
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a integer with [name] as the parameter name.
+ * @param min the minimum value.
+ * @param max the maximum value.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.integer(
+    name: String,
+    min: Int = -Int.MAX_VALUE,
+    max: Int = Int.MAX_VALUE,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, Int>(
+        name,
+        IntegerArgumentType.integer(min, max)
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a long with [name] as the parameter name.
+ * @param min the minimum value.
+ * @param max the maximum value.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.long(
+    name: String,
+    min: Long = -Long.MAX_VALUE,
+    max: Long = Long.MAX_VALUE,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, Long>(
+        name,
+        LongArgumentType.longArg(min, max)
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds an executor to the command tree.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+@JvmName("executesReturn")
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.executes(
+    action: CommandActionReturn<S>
+) {
+    executes(action)
+}
+
+/**
+ * Adds an executor to the command tree.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.executes(
+    action: CommandAction<S>
+) {
+    executes {
+        try {
+            it.action()
+            1
+        } catch (error: Throwable) {
```

Comment:
Euh, this is not what i'd expect. If the function throws a `CommandSyntaxException` it should rethrow that. If it's any other error, idk if it even should be caught, and if it is, it should be wrapped and rethrown as a `CommandSyntaxException` but that feels weird.